src/utils/workBannerUtility.ts

```diff
@@ -0,0 +1,26 @@
+import { WorkBannerConsumerProps } from 'HOCs/WorkBannerProvider';
+
+export const ACCOUNT_VERIFICATION = 'Click here to verify your email and phone number to start booking!';
+
+export const showAccountVerificationBanner = async (
+  isUserVerified: boolean,
+  bannerDispatch: WorkBannerConsumerProps['bannerDispatch'],
+  bannerState: WorkBannerConsumerProps['bannerState']): Promise<void> => {
+  const userVerified = isUserVerified && bannerState.show;
+  const userUnverified = !isUserVerified && !bannerState.show;
+  const options = {
+    message: ACCOUNT_VERIFICATION,
+    background: 'secondary',
+    textColor: 'white',
+    to: '/account/verification',
+  }
+
+  if (userVerified) {
+    await bannerDispatch({ type: 'close' });
+  }
+
+  if (userUnverified) {
+    await bannerDispatch({ type: 'set', payload: options });
+    await bannerDispatch({ type: 'open' });
```

Comment:
doesn't this happen synchronously without `await`

Reply:
or maybe like `this.setState` it might batch the dispatch changes together

Reply:
That's a good question.  Originally we used the await in case this because a race condition on whether the banner would set banner props first or open first.  I think in the background `useReducer` still uses setState, so this might still hold true.  An option we can do is wrap the dispatch calls with `new Promise` but I think that might add unnecessary bulk.  We can try to remove it for now and see if the banner every loads before the message.  Thoughts?

Reply:
ya sounds good to me 👍